server/shared/publishing/helpers.js

```diff
@@ -151,23 +151,39 @@ function publishAssessments(parent) {
 
 function fetchContainers(parent) {
   const typeConfigs = getSupportedContainers(parent.type);
-  const isCore = it => !containerRegistry.getStaticsResolver(it.type);
-  const coreTypes = typeConfigs.filter(isCore).map(it => it.type);
+  const { defaults, custom } = getGroupedContainerTypes(typeConfigs);
 
-  return Promise.all([
-    parent.getChildren({ where: { type: coreTypes } }).map(fetchDefaultContainer),
-    fetchCustomContainers(parent)
-  ])
-  .reduce((containers, groupedContainers) => {
+  const fetches = [];
+  if (defaults.length) fetches.push(fetchDefaultContainers(parent, defaults));
+  if (custom.length) fetches.push(fetchCustomContainers(parent, custom));
+  if (!fetches.length) return [];
+  return Promise.all(fetches).reduce((containers, groupedContainers) => {
     const mappedContainers = groupedContainers.map(it => {
       const config = find(typeConfigs, { type: it.type });
       const publishedAs = get(config, 'publishedAs', 'container');
-      return { ...it, publishedAs };
+      const { templateId } = config;
+      return { ...it, templateId, publishedAs };
     });
     return containers.concat(mappedContainers);
   }, []);
 }
 
+function getGroupedContainerTypes(typeConfigs) {
+  return typeConfigs.reduce((acc, { type, templateId }) => {
+    if (containerRegistry.getContentFetcher(templateId)) {
+      acc.custom.push(templateId);
+    } else {
+      acc.defaults.push(type);
+    }
+    return acc;
+  }, { defaults: [], custom: [] });
+}
+
+function fetchDefaultContainers(parent, types) {
```

Comment:
See comment below.

Reply:
I see that you commented this one as well. They way we should do this if we want the types fetched in these methods is to build a type look up on server start in the `config/shared/activities.js` file. This is something that should probably be done no matter what path we take.

Reply:
We don't really do what wrote anywhere (as far as I can tell). So for now, I will remove the finding of the container types from the `fetchContainers` method and move it to the sub-methods. And then we can see if we want to go in another way.